### PR TITLE
Werkzeug/Flask 3 compatibility

### PR DESCRIPTION
**Pull request by frantisekz**
This is still WIP (Session stuff is remaining, maybe some more stuff after it).

### CODE REVIEW

**Comment by praiskup** [comment]
Nice, thank you for the help with this.

**Comment by frantisekz** [comment]
> Nice, thank you for the help with this.

You're welcome :)

So far, got to "============ 22 failed, 456 passed, 25 warnings, 1 error in 21.44s ============="

**Comment by frantisekz** [comment]
Okay, unit tests are passing, so... Ready to review :)

(It needs the following in the Fedora to build: https://bugzilla.redhat.com/show_bug.cgi?id=2255280 )

**Comment by FrostyX** [comment]
Thank you very much for the update. Is it fair to set the `blocked` label for this PR and wait with merging till the `python-flask-session` is added to Fedora?

In the meantime, I took the package review and some comments.

**Comment by xsuchy** [comment]
Error: PYLINT_WARNING:
coprs_frontend/coprs/__init__.py:49:4: W0621[redefined-outer-name]: db_session_scope: Redefining name 'session' from outer scope (line 42)

The variable on the line 49 should be renamed.

**Comment by frantisekz** [comment]
> Error: PYLINT_WARNING: coprs_frontend/coprs/**init**.py:49:4: W0621[redefined-outer-name]: db_session_scope: Redefining name 'session' from outer scope (line 42)
> 
> The variable on the line 49 should be renamed.

Should be addressed now.

**Comment by frantisekz** [comment]
The new linting errors seem to be unrelated to this change?

File: frontend/coprs_frontend/tests/test_exceptions.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 class TestExceptionHandling(CoprsTestCase):
     def test_json_only_for_api(self):
         app.config["SESSION_COOKIE_DOMAIN"] = "localhost.localdomain"
+        app.config["SESSION_TYPE"] = "redis"
```
Comment (praiskup) [inline frontend/coprs_frontend/tests/test_exceptions.py:14]: Can you document why this is needed?

File: frontend/coprs_frontend/coprs/__init__.py [hunk 2]
```diff
@@ -38,17 +39,19 @@
     extension_responses=[TeamsResponse]
 )
 
+session = Session(app)
```
Comment (praiskup) [inline frontend/coprs_frontend/coprs/__init__.py:42]: How is this global used?

Comment (frantisekz) [reply frontend/coprs_frontend/coprs/__init__.py:42]: It's not, sorry, in my projects, I usually declare these flask.app wrappers as variables even if they're not used at all. Can be safely changed to Session(app).

File: frontend/coprs_frontend/tests/test_exceptions.py (hunk 1 above)
Comment (frantisekz) [reply frontend/coprs_frontend/tests/test_exceptions.py:14]: This is leftover from debugging, not needed.

File: frontend/coprs_frontend/coprs/__init__.py (hunk 2 above)
Comment (praiskup) [reply frontend/coprs_frontend/coprs/__init__.py:42]: I'm fine with this, just wanted to confirm that.

**Review by praiskup — APPROVED** [review]
Thank you!

**Pull request merged**